### PR TITLE
Task: upvote and downvote on issues

### LINKED ISSUE #1101: upvote and downvote on issues (show like counts, and add a dislike and show counts) (different than flag)
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Additional context**

### PR DESCRIPTION
**Pull request by thedudeontitan**
fixes #1101 and part of #1627 

Added downvote button with count for upvote and downvotes and removed the redundant image icon.

image

### CODE REVIEW

**Review by DonnieBLT — DISMISSED** [review]

**Comment by thedudeontitan** [comment]
Hey, fixed the pre commit check.

File: website/views.py [hunk 1]
```diff
@@ -3591,6 +3590,30 @@ def like_issue2(request, issue_pk):
     context["likes"] = total_votes
     return render(request, "includes/_likes2.html", context)
 
+@login_required(login_url="/accounts/login")
+def dislike_issue2(request, issue_pk):
+    context = {}
+    issue_pk = int(issue_pk)
+    issue = Issue.objects.get(pk=issue_pk)
+    userprof = UserProfile.objects.get(user=request.user)
+    if userprof in UserProfile.objects.filter(issue_downvoted=issue):
```
Comment (arkid15r) [inline website/views.py:3599]: I suggest using `UserProfile.objects.filter(issue_downvoted=issue, user=request.user).exists()` instead.

File: website/views.py [hunk 2]
```diff
@@ -3591,6 +3590,30 @@ def like_issue2(request, issue_pk):
     context["likes"] = total_votes
     return render(request, "includes/_likes2.html", context)
 
+@login_required(login_url="/accounts/login")
+def dislike_issue2(request, issue_pk):
+    context = {}
+    issue_pk = int(issue_pk)
+    issue = Issue.objects.get(pk=issue_pk)
```
Comment (arkid15r) [inline website/views.py:3597]: What if issue with issue_pk doesn't exist?

File: website/views.py [hunk 3]
```diff
@@ -3591,6 +3590,30 @@ def like_issue2(request, issue_pk):
...
+    userprof = UserProfile.objects.get(user=request.user)
+    if userprof in UserProfile.objects.filter(issue_downvoted=issue):
+        userprof.issue_downvoted.remove(issue)
+    else:
+        userprof.issue_downvoted.add(issue)
+    userprof.save()
```
Comment (arkid15r) [inline website/views.py:3603]: Do you need `save()` here?

File: website/views.py [hunk 4]
```diff
@@ -3591,6 +3590,30 @@ def like_issue2(request, issue_pk):
...
+    total_votes = UserProfile.objects.filter(issue_downvoted=issue).count()
+    context["object"] = issue
+    context["dislikes"] = total_votes
+    return render(request, "includes/_dislike2.html", context)
+
+@login_required(login_url="/accounts/login")
+def vote_count(request, issue_pk):
+    issue_pk = int(issue_pk)
+    issue = Issue.objects.get(pk=issue_pk)
```
Comment (arkid15r) [inline website/views.py:3612]: Same here -- add issue existence check.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
please see recent feedback thanks @arkid15r

**Comment by thedudeontitan** [comment]
Hey, made the suggested changes please check.

File: website/views.py (hunk 3 above)
Comment (arkid15r) [reply website/views.py:3609]: I don't think you need it. As long as you update the object's m2m relations you don't need to do explicit `save()` for the object itself.

https://docs.djangoproject.com/en/5.0/topics/db/examples/many_to_many/#many-to-many-relationships

**Review by arkid15r — CHANGES_REQUESTED** [review]
Could you take another look?

File: website/views.py (hunk 3 above)
Comment (thedudeontitan) [reply website/views.py:3609]: Should i remove save() from both the views, i.e. like_issue2() and dislike_issue2()?

Comment (arkid15r) [reply website/views.py:3609]: It'd be great from consistency perspective.

**Comment by thedudeontitan** [comment]
Thanks for the suggestion! Have pushed new changes please check

File: website/templates/includes/_dislike2.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{% load gravatar %}
```
Comment (arkid15r) [inline website/templates/includes/_dislike2.html:1]: Do you use this tag anywhere?

**Review by arkid15r — COMMENTED** [review]
Python stuff LGTM

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**